addons/block_code/blocks/logic/else_if.tres

```diff
@@ -0,0 +1,15 @@
+[gd_resource type="Resource" load_steps=2 format=3 uid="uid://by53vmmn3wtny"]
+
+[ext_resource type="Script" path="res://addons/block_code/code_generation/block_definition.gd" id="1_kgjks"]
+
+[resource]
+script = ExtResource("1_kgjks")
+name = &"else_if"
+type = 4
+variant_type = 0
+display_template = "Else if {condition: BOOL}"
+code_template = "elif {condition}:"
```

Comment:
@wnbaum I wonder why you decided to make "else" or "else if" separate blocks.
![Captura desde 2024-08-06 07-51-51](https://github.com/user-attachments/assets/d9770b9a-1242-4198-8635-025dcb6f3cfe)

This can create scripts with errors:
![Captura desde 2024-08-06 08-05-54](https://github.com/user-attachments/assets/7727621d-299c-4c67-8857-63053f7406b7)

Is it because blocks with multiple slots are hard in this new architecture? In that case it is less flexible than what we had, going against the flexibility goal.

IMO the beauty of blocks is that they prevent parse and syntax errors.

Reply:
Yes, the main reason was ease of implementation. We could accomplish this with different `BlockDefinition` types, like `StatementBlockDefinition` and `ControlBlockDefinition` as well as a more complex abstract syntax tree. As you can see in the `BlockDefinition` class we have a `block_type` and a `variant_type` which lets us combine all the current block types into one data model. This is nice for simplicity, but only allows one statement for control blocks (unless we were to generalize the model even more). 

Also, the AST is currently made up of nodes with children. Each entry block has children, and each control block has children, and the children are the sequence of blocks that are indented under the current statement. We would need to expand the AST, BlockNameTree, and UI generation functions to take into account multiple children groups per node. This wasn't an issue before since we just saved the paths to each block, and generated the code through the UI, which IMO was very ugly.

So it's definitely possible to do undo this regression, but it would add a lot more to this PR. Also, there are some benefits to having the control blocks being like this. If you wanted to convert an `if/else` statement to an `if` in the old version, you would have to take all the blocks out, switch out the control block, then put all the blocks back in. Whereas with this you can just chuck the `else` block. The only thing you would have to do to guarantee correct scripts is make sure `else` and `elif` blocks can only be attached to `if` and `elif` blocks.

Anyways, there's definitely arguments for both UI methods, so I thought it would be best to do the simplest implementation and then decide if it is really worth porting back later. IIRC @dbnicholson had similar thoughts when we talked about it.